### PR TITLE
GameVersion and ConverterVersion

### PR DESCRIPTION
**Pull request by Zemurin**
Phase 2:
- GameVersion now knows how to extract itself from installations (with more checks) + related tests on live files
- ConverterVersion is now updated and generalized, and generates own output dynamically.

### CODE REVIEW

**Comment by Zemurin** [comment]
Codacy error is intentional.

File: GameVersion.h [hunk 1]
```diff
@@ -50,6 +50,10 @@ class GameVersion: commonItems::convenientParser
 
 	friend std::ostream& operator<<(std::ostream&, const GameVersion& version);
 
+	[[nodiscard]] static std::optional<GameVersion> extractLauncherVersion(const std::string& filePath); // modern PDX games, scrapes launcher-settings.json.
+	[[nodiscard]] static std::optional<GameVersion> extractReadMeVersion(const std::string& filePath); // extracts version from Vic2 ReadMe.txt/Readme.txt.
+	[[nodiscard]] static std::optional<GameVersion> extractChangeLogVersion(const std::string& filePath); // extracts version from CK2 ChangeLog.txt.
+
```
Comment (IhateTrains) [inline GameVersion.h:56]: Maybe rename these to `extractVersionFromX`? `extractLauncherVersion` sounds like you're checking the launcher's version.

Comment (Zemurin) [reply GameVersion.h]: yup yup

File: GameVersion.cpp [hunk 2]
```diff
@@ -347,3 +348,170 @@ bool GameVersion::isLargerishThan(const GameVersion& rhs) const
 
 	return true;
 }
+
+std::optional<GameVersion> GameVersion::extractLauncherVersion(const std::string& filePath)
```
Comment (IhateTrains) [inline GameVersion.cpp]: Have you considered using a JSON library? I can recommend nlhomann one, comes in a single header form.

Comment (IhateTrains) [inline GameVersion.cpp]: Have you considered using a JSON library? I can recommend nlhomann one, comes in a single header form.

**Review by IhateTrains — APPROVED** [review]
Solid work

File: GameVersion.cpp (hunk 2 above)
Comment (Zemurin) [reply GameVersion.cpp]: I did and I oped for a simpler straight scrape. We only need a single value, not actual json-to-something support.

**Review by Idhrendur — APPROVED** [review]

**Pull request merged**